common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -683,6 +683,16 @@ public UTF8String trimRight() {
     return copyUTF8String(0, e);
   }
 
+  /**
+   * Return the length of string stripped the tailing space after `limit`.
+   */
+  public int canTrimRightTo(int numOfSpaces) {
```

Comment:
it's a bit weird that a method with the name `canTrimRightTo` returns int not boolean.

Reply:
renamed to `effectiveLength`m WDYT?